pkg/util/sets/string.go

```diff
@@ -14,44 +14,47 @@
 
 package sets
 
-import "sort"
+import (
+	"golang.org/x/exp/constraints"
+	"golang.org/x/exp/slices"
+)
 
-type Set map[string]struct{}
+type Set[T constraints.Ordered] map[T]struct{}
 
 // NewWithLength returns an empty Set with the given capacity.
 // It's only a hint, not a limitation.
-func NewWithLength(l int) Set {
-	return make(Set, l)
+func NewWithLength[T constraints.Ordered](l int) Set[T] {
+	return make(Set[T], l)
 }
 
 // New creates a new Set with the given items.
-func New(items ...string) Set {
-	s := NewWithLength(len(items))
+func New[T constraints.Ordered](items ...T) Set[T] {
```

Comment:
This is making the usage more complex, as a fundamental lib , better ux is more important than dedup.

Reply:
I agree to an extent with this. The usage certainly is more complex compared to earlier.

Reply:
I agree to an extent with this. The usage certainly is more complex compared to earlier.

Reply:
I agree to an extent with this. The usage certainly is more complex compared to earlier.

Reply:
Its _slightly_ more complex for strings - but there are 100s of places where we _should_ be using Sets throughout the code but do not because it isn't generic.

If the concern is `sets.Set` -> `sets.Set[string]` is verbose I can add `type String = Set[string]` and then you can use it like `sets.String`?

Reply:
That seems like a good idea for me for readability.